src/browser/BrowserService.cpp

```diff
@@ -790,6 +831,75 @@ Database* BrowserService::getDatabase()
     return nullptr;
 }
 
+bool BrowserService::moveSettingsToCustomData(Entry* entry, const QString& name) const
+{
+    if (entry->attributes()->contains(name)) {
+        QString attr = entry->attributes()->value(name);
+        entry->beginUpdate();
+        if (!attr.isEmpty()) {
+            entry->customData()->set(KEEPASSXCBROWSER_NAME, attr);
+        }
+        entry->attributes()->remove(name);
+        entry->endUpdate();
+        return true;
+    }
+    return false;
+}
+
+int BrowserService::moveKeysToCustomData(Entry* entry, Database* db) const
+{
+    int keyCounter = 0;
+    for (const auto& key : entry->attributes()->keys()) {
+        if (key.contains(ASSOCIATE_KEY_PREFIX)) {
+            QString publicKey = key;
+            publicKey.remove(ASSOCIATE_KEY_PREFIX);
+
+            // Add key to database custom data
+            if (db && !db->metadata()->customData()->contains(QLatin1String(ASSOCIATE_KEY_PREFIX) + publicKey)) {
+                db->metadata()->customData()->set(QLatin1String(ASSOCIATE_KEY_PREFIX) + publicKey, entry->attributes()->value(key));
+                ++keyCounter;
+            }
+        }
+    }
+
+    return keyCounter;
+}
+
+bool BrowserService::checkLegacySettings()
+{
+    Database* db = m_dbTabWidget->currentDatabaseWidget()->database();
+    if (!db) {
+        return false;
+    }
+
+    bool legacySettingsFound = false;
+    QList<Entry*> entries = db->rootGroup()->entriesRecursive();
+    for (const auto& e : entries) {
+        if ((e->attributes()->contains(KEEPASSHTTP_NAME) || e->attributes()->contains(KEEPASSXCBROWSER_NAME)) ||
+            (e->title() == KEEPASSHTTP_NAME || e->title() == KEEPASSXCBROWSER_NAME)) {
+            legacySettingsFound = true;
+            break;
+        }
+    }
+
+    if (!legacySettingsFound) {
+        return false;
+    }
+
+    auto dialogResult = QMessageBox::warning(nullptr,
+                                             tr("KeePassXC: Legacy browser integration settings detected"),
+                                             tr("Legacy browser integration settings have been detected.\n"
+                                                "Do you want to transfer the settings and stored keys to custom data?\n"
```

Comment:
I would just say "Do you want to upgrade the settings to the latest standard?\nThis is necessary to maintain compatibility with the browser plugin."